examples/react/start-workos/app/routes/__root.tsx

```diff
@@ -0,0 +1,100 @@
+import { Box, Button, Card, Container, Flex, Theme } from '@radix-ui/themes';
+import '@radix-ui/themes/styles.css';
+import { HeadContent, Link, Outlet, Scripts, createRootRoute } from '@tanstack/react-router';
+import { TanStackRouterDevtools } from '@tanstack/react-router-devtools';
+import { Suspense } from 'react';
+import { getAuth, getSignInUrl } from '../authkit/serverFunctions';
+import Footer from '../components/footer';
+import SignInButton from '../components/sign-in-button';
+import type { ReactNode } from 'react';
+
+export const Route = createRootRoute({
+  head: () => ({
+    meta: [
+      {
+        charSet: 'utf-8',
+      },
+      {
+        name: 'viewport',
+        content: 'width=device-width, initial-scale=1',
+      },
+      {
+        title: 'AuthKit Example in TanStack Start',
+      },
+    ],
+  }),
+  beforeLoad: async () => {
+    const { user } = await getAuth();
```

Comment:
Are y'all happy with `beforeLoad`, on the `rootRoute`, being called on _every_ navigation?

Is this perhaps something TanStack Query would be better for? or is this the intention?